argilla/src/argilla/_helpers/_datasets_server.py

```diff
@@ -0,0 +1,175 @@
+    Extract the features from the response and return them as a dictionary.
+
+    Parameters:
+        repo_id (str): The repository ID of the dataset.
+        config (str, optional): The configuration of the dataset. Defaults to None.
+
+    Returns:
+        Dict[str, Dict[str, Any]]: The features of the dataset.
+    """
+
+    url = f"{DATASETS_SERVER_BASE_URL}/info?dataset={repo_id}"
+
+    try:
+        response = requests.get(url, headers=DATASETS_SERVER_HEADERS, timeout=DATASETS_SERVER_TIMEOUT)
+    except requests.exceptions.RequestException as e:
+        raise DatasetsServerException() from e
+
+    try:
+        response.raise_for_status()
+    except requests.exceptions.HTTPError as e:
+        raise DatasetsServerException() from e
+
+    try:
+        response_json = response.json()
+    except Exception as e:
+        raise DatasetsServerException() from e
+
+    try:
+        dataset_info = response_json["dataset_info"]
+    except KeyError as e:
+        raise DatasetsServerException() from e
+
+    available_configs = list(dataset_info.keys())
+
+    if len(available_configs) > 1 and config is None:
+        raise ValueError("Multiple configurations found. Please specify a config.")
+    elif len(available_configs) == 1:
+        config = available_configs[0]
+    elif config not in available_configs:
+        raise ValueError(f"Configuration '{config}' not found.")
+
+    try:
+        features = dataset_info[config]["features"]
+    except KeyError as e:
+        raise DatasetsServerException() from e
+
+    return features
+
+
+def _map_feature_type(feature):
+    """Map the feature type to the corresponding FeatureType enum."""
+
+    if isinstance(feature, list) and len(feature) > 0 and isinstance(feature[0], dict):
+        sub_features = feature[0]
+        if _is_chat_feature(sub_features):
+            return FeatureType.CHAT
+
+    hf_type = feature.get("_type")
+    dtype = feature.get("dtype")
+
+    if hf_type == "Value":
+        if dtype == "string":
+            return FeatureType.TEXT
+        elif dtype in ["int32", "int64"]:
+            return FeatureType.INT
+        elif dtype in ["float32", "float64"]:
+            return FeatureType.FLOAT
+    elif hf_type == "Image":
+        return FeatureType.IMAGE
+    elif hf_type == "ClassLabel":
+        return FeatureType.LABEL
+    else:
+        warnings.warn(f"Unsupported feature type. hf_type: {hf_type}, dtype: {dtype}")
+
+
+def _is_chat_feature(sub_features):
+    """Check if the sub_features correspond to an rg.ChatField."""
+    return (
+        "content" in sub_features
+        and "role" in sub_features
+        and sub_features["content"]["_type"] == "Value"
+        and sub_features["role"]["_type"] == "Value"
+        and sub_features["content"].get("dtype") == "string"
+        and sub_features["role"].get("dtype") == "string"
+    )
+
+
+def _define_settings_from_features(features: Union[List[Dict], Dict[str, Any]], config=None) -> "Settings":
```

Comment:
Regarding this comment https://github.com/argilla-io/argilla/pull/5472/files#r1751978426
```suggestion
def _define_settings_from_features(features: Union[List[Dict], Dict[str, Any]]) -> "Settings":
```

Reply:
I think the `config` argument is not used on this function. We can remove it.